### PR TITLE
tidb: Add time format description

### PR DESCRIPTION
**Pull request by 7yyo**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

The insert of TiDB does not support `yyyymmdd hhmmss` or `yyyymmdd hh:mm:ss` format, consistent with the behavior of `mysql`. But this format may appear in user case ( `Oracle` behavior ). Because there are corresponding comments in the source code, I think it's better to add it to the official doc.

https://github.com/pingcap/tidb/blob/master/types/time.go#L1919

E.g
```mysql
mysql> insert into t values('20201111 111111');
ERROR 1292 (22007): Incorrect datetime value: '20201111 111111' for column 'd' at row 1
```

```mysql
mysql> insert into t values('20201111 11:11:11');
ERROR 1292 (22007): Incorrect datetime value: '20201111 11:11:11' for column 'd' at row 1
```

```mysql
mysql> insert into t values('2020-11-11 111111');
ERROR 1292 (22007): Incorrect datetime value: '2020-11-11 111111' for column 'd' at row 1
```

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: literal-values.md [hunk 1]
```diff
@@ -96,7 +96,7 @@ Date 跟 Time 字面值有几种格式，例如用字符串表示，或者直接
 
 TiDB 的 Date 值有以下几种格式：
 
-* `'YYYY-MM-DD'` 或者 `'YY-MM-DD'`，这里的 `-` 分隔符并不是严格的，可以是任意的标点符号。比如 `'2017-08-24'`，`'2017&08&24'`， `'2012@12^31'` 都是一样的。唯一需要特别对待的是 '.' 号，它被当做是小数点，用于分隔整数和小数部分。
+* `'YYYY-MM-DD'` 或者 `'YY-MM-DD'`，这里的 `-` 分隔符并不是严格的，可以是任意的标点符号。比如 `'2017-08-24'`，`'2017&08&24'`， `'2012@12^31'` 都是一样的。唯一需要特别对待的是 '.' 号，它被当做是小数点，用于分隔整数和小数部分。缺少分隔符的情况， 必须使用 `YYYYMMDDHHMMSS` 或 `YYMMDDHHMMSS` 格式，例如 `YYYYMMDD HHMMSS` ， `YYYYMMDD HH:MM:DD`， `YYYY-MM-DD HHMMSS`会插入失败。
```
Comment (TomShawn) [inline literal-values.md:99]:
```suggestion
* `'YYYY-MM-DD'` 或者 `'YY-MM-DD'`，这里的 `-` 分隔符并不是严格的，可以是任意的标点符号。比如 `'2017-08-24'`，`'2017&08&24'`， `'2012@12^31'` 都是一样的。唯一需要特别对待的是 '.' 号，它被当做是小数点，用于分隔整数和小数部分。缺少分隔符的情况， 必须使用 `YYYYMMDDHHMMSS` 或 `YYMMDDHHMMSS` 格式。例如 `YYYYMMDD HHMMSS` 、 `YYYYMMDD HH:MM:DD`、 `YYYY-MM-DD HHMMSS` 等格式的值会插入失败。
```

Comment (XuHuaiyu) [reply literal-values.md:99]: 103 行已经列出来了这 2 个格式，为什么这里需要再单独列一下？

**Comment by TomShawn** [comment]
@7yyo 能否将此说明移到 103 行，只保留 “例如 `YYYYMMDD HHMMSS` 、 `YYYYMMDD HH:MM:DD`、 `YYYY-MM-DD HHMMSS` 会插入失败。”？

**Comment by 7yyo** [comment]
> @7yyo 能否将此说明移到 103 行，只保留 “例如 `YYYYMMDD HHMMSS` 、 `YYYYMMDD HH:MM:DD`、 `YYYY-MM-DD HHMMSS` 会插入失败。”？

done

File: literal-values.md [hunk 2]
```diff
@@ -100,7 +100,7 @@ TiDB 的 Date 值有以下几种格式：
 
     Date 和 Time 部分可以被 'T' 分隔，它的作用跟空格符是一样的，例如 `2017-8-24 10:42:00` 跟 `2017-8-24T10:42:00` 是一样的。
 
-* `'YYYYMMDDHHMMSS'` 或者 `'YYMMDDHHMMSS'`，例如 `'20170824104520'` 和 `'170824104520'` 被当做是 `'2017-08-24 10:45:20'`，但是如果你提供了一个超过范围的值，例如`'170824304520'`，那这就不是一个有效的 Date 字面值。
+* `'YYYYMMDDHHMMSS'` 或者 `'YYMMDDHHMMSS'`，例如 `'20170824104520'` 和 `'170824104520'` 被当做是 `'2017-08-24 10:45:20'`，但是如果你提供了一个超过范围的值，例如`'170824304520'`，那这就不是一个有效的 Date 字面值。需要注意`YYYYMMDD HHMMSS` ， `YYYYMMDD HH:MM:DD`， `YYYY-MM-DD HHMMSS`会插入失败
```
Comment (TomShawn) [inline literal-values.md:103]:
```suggestion
* `'YYYYMMDDHHMMSS'` 或者 `'YYMMDDHHMMSS'`，例如 `'20170824104520'` 和 `'170824104520'` 被当做是 `'2017-08-24 10:45:20'`，但是如果你提供了一个超过范围的值，例如`'170824304520'`，那这就不是一个有效的 Date 字面值。需要注意 `YYYYMMDD HHMMSS` ，`YYYYMMDD HH:MM:DD`，`YYYY-MM-DD HHMMSS` 等不正确的格式会插入失败。
```

**Pull request merged**